File: HoTT-Model/src/main/java/gde/model/winged/WingedPhase.java
/**
 *  HoTT Transmitter Config
 *  Copyright (C) 2013  Oliver Treichel
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package gde.model.winged;

import java.util.Arrays;

import javax.xml.bind.annotation.XmlElementWrapper;

import gde.model.Curve;
import gde.model.Phase;

/**
 * @author oli
 *
 */
public class WingedPhase extends Phase {
	private static final long serialVersionUID = 1L;

	private Curve brakeElevatorCurve;
	private WingedMixer[] brakeMixer;
	private WingedMixer[] multiFlapMixer;
	private WingedMixer[] wingMixer;
	private WingedTrim wingTrim;

	@Override
	public boolean equals(final Object obj) {
		if (this == obj) {
			return true;
		}
		if (!super.equals(obj)) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final WingedPhase other = (WingedPhase) obj;
		if (brakeElevatorCurve == null) {
			if (other.brakeElevatorCurve != null) {
				return false;
			}
		} else if (!brakeElevatorCurve.equals(other.brakeElevatorCurve)) {
			return false;
		}
		if (!Arrays.equals(brakeMixer, other.brakeMixer)) {
			return false;
		}
		if (!Arrays.equals(multiFlapMixer, other.multiFlapMixer)) {
			return false;
		}
		if (!Arrays.equals(wingMixer, other.wingMixer)) {
			return false;
		}
		if (wingTrim == null) {
			if (other.wingTrim != null) {
				return false;
			}
		} else if (!wingTrim.equals(other.wingTrim)) {
			return false;
		}
		return true;
	}

	public Curve getBrakeElevatorCurve() {
		return brakeElevatorCurve;
	}

	@XmlElementWrapper(name = "brakeMixers")
	public WingedMixer[] getBrakeMixer() {
		return brakeMixer;
	}

	@XmlElementWrapper(name = "multiFlapMixers")
	public WingedMixer[] getMultiFlapMixer() {
		return multiFlapMixer;
	}

	@XmlElementWrapper(name = "wingMixers")
	public WingedMixer[] getWingMixer() {
		return wingMixer;
	}

	public WingedTrim getWingTrim() {
		return wingTrim;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = super.hashCode();
		result = prime * result + (brakeElevatorCurve == null ? 0 : brakeElevatorCurve.hashCode());
		result = prime * result + Arrays.hashCode(brakeMixer);
		result = prime * result + Arrays.hashCode(multiFlapMixer);
		result = prime * result + Arrays.hashCode(wingMixer);
		result = prime * result + (wingTrim == null ? 0 : wingTrim.hashCode());
		return result;
	}

	public void setBrakeElevatorCurve(final Curve brakeElevatorCurve) {
		this.brakeElevatorCurve = brakeElevatorCurve;
	}

	public void setBrakeMixer(final WingedMixer[] brakeMixer) {
		this.brakeMixer = brakeMixer;
	}

	public void setMultiFlapMixer(final WingedMixer[] multiFlapMixer) {
		this.multiFlapMixer = multiFlapMixer;
	}

	public void setWingMixer(final WingedMixer[] wingMixer) {
		this.wingMixer = wingMixer;
	}

	public void setWingTrim(final WingedTrim wingTrim) {
		this.wingTrim = wingTrim;
	}
}

File: HoTT-Model/src/main/java/de/treichels/hott/model/winged/WingedPhase.kt
/**
 * HoTT Transmitter Config Copyright (C) 2013 Oliver Treichel
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with this program. If not, see <http:></http:>//www.gnu.org/licenses/>.
 */
package de.treichels.hott.model.winged

import de.treichels.hott.model.Curve
import de.treichels.hott.model.Phase
import javax.xml.bind.annotation.XmlElementWrapper

/**
 * @author Oliver Treichel &lt;oli@treichels.de&gt;
 */
class WingedPhase(
        var brakeElevatorCurve: Curve = Curve(),
        @get:XmlElementWrapper(name = "brakeMixers")
        var brakeMixer: List<WingedMixer> = listOf(),
        @get:XmlElementWrapper(name = "multiFlapMixers")
        var multiFlapMixer: List<WingedMixer> = mutableListOf(),
        @get:XmlElementWrapper(name = "wingMixers")
        var wingMixer: List<WingedMixer> = mutableListOf(),
        var wingTrim: WingedTrim = WingedTrim()
) : Phase() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is WingedPhase) return false
        if (!super.equals(other)) return false

        if (brakeElevatorCurve != other.brakeElevatorCurve) return false
        if (brakeMixer != other.brakeMixer) return false
        if (multiFlapMixer != other.multiFlapMixer) return false
        if (wingMixer != other.wingMixer) return false
        if (wingTrim != other.wingTrim) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + brakeElevatorCurve.hashCode()
        result = 31 * result + brakeMixer.hashCode()
        result = 31 * result + multiFlapMixer.hashCode()
        result = 31 * result + wingMixer.hashCode()
        result = 31 * result + wingTrim.hashCode()
        return result
    }
}
